packages/core/src/state-management/repository.ts

```diff
@@ -287,7 +287,7 @@ export class Repository {
   }
 
   pin(streamStateHolder: StreamStateHolder): Promise<void> {
-    return this.#deps.pinStore.add(streamStateHolder)
+    return this.#deps.pinStore.add(new RunningState(streamStateHolder.state, StateSource.NETWORK))
```

Comment:
this definitely isn't safe.  The StreamStateHolder likely *is* a RunningState, and you're discarding the information from that RunningState about its StateSource, as well as breaking any rxjs pipelines built on the original RunningState.

The thing to do here is probably to update the signature of `pin` to be more specific and specify that its input is a  `RunningState` instance instead of a `StreamStateHolder`

Reply:
Yea that was actually my first thought but I went with this tactic so there would be minimal changes. Thank you clarifying the problem with this tactic! I'll try updating the signature again and if it touches too many things i'll let you know